src/transformStateWithClones.js

```diff
@@ -8,6 +8,32 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  const stateCopy = { ...state };
+
+  for (const a of actions) {
+    switch (a.type) {
+      case 'addProperties':
+        for (const d in a.extraData) {
+          stateCopy[d] = a.extraData[d];
+        }
```

Comment:
Instead of loop here, you could do this:
```suggestion
  Object.assign(stateCopy, a.extraData);
```